### PR TITLE
Show message when checking for updates offline

### PR DESCRIPTION
**Pull request by sharon-wang**
### Intent

Issue: https://github.com/rstudio/rstudio-pro/issues/3388 (not specifically a pro issue).

The following dialog will show if a user tries to manually check for updates, but an error occurs during the download (user is offline):
> image

### Approach

Previously, RStudio would report that the user is up-to-date when manually checking for updates offline. Now, the user will see a dialog to inform them that a connection error occurred and to check via the downloads page directly.

Note that this dialog will not be triggered by automated checks for updates (eg. at startup), only when the user manually checks.

#### Steps
1. Turn off WiFi
2. While offline, open RStudio
3. Click on `Help` -> `Check for Updates`
4. A dialog should show, which asks the user to check the downloads site for updates

The dialog should not be triggered by automatic update checks that occur at IDE startup.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants_fr.properties [hunk 1]
```diff
@@ -154,3 +154,4 @@ saveNoLabel=Ne pas sauvegarder
 consolePaneName=Console
 activeText=(active)
 requestLogVisualization=<p>Cliquez sur une demande pour voir les détails. Cliquez sur l''arrière-plan pour afficher à nouveau ces instructions.</p><h4>Commandes disponibles:</h4><ul><li>Esc: Fermer</li><li>P: Lecture/pause</li><li>E: Exporter</li><li>I: Importer</li><li>+/-: Zoom avant/arrière</li></ul>
+visitWebsiteForNewVersionText=Veuillez visiter https://posit.co/download/rstudio-desktop/ pour vérifier si une nouvelle version est disponible.
```
Comment (sharon-wang) [inline src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants_fr.properties:157]: @romainfrancois I used a translator to generate this -- could you please confirm if it's correct?

File: src/cpp/session/modules/SessionUpdates.cpp [hunk 2]
```diff
@@ -134,7 +134,14 @@ void endRPCUpdateCheck(const json::JsonRpcFunctionContinuation& cont,
 {
    json::JsonRpcResponse response;
    response.setResult(jsonFromProcessResult(result));
-   cont(Success(), &response);
+   if (result.stdErr.empty()) {
```
Comment (gtritchie) [inline src/cpp/session/modules/SessionUpdates.cpp:137]: Nicpick: Braces on separate line (our Java/C++ style guidelines: https://github.com/rstudio/rstudio/wiki/Coding-Standards-(java,-cpp))

File: src/gwt/src/org/rstudio/studio/client/application/ui/impl/DesktopApplicationHeader.java [hunk 3]
```diff
@@ -447,9 +447,14 @@ public void onResponseReceived(UpdateCheckResult result)
          @Override
          public void onError(ServerError error)
          {
-            globalDisplay_.showErrorMessage(constants_.errorCheckingUpdatesMessage(),
+            // Only show the error message when manually checking for updates 
+            if (manual) {
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/impl/DesktopApplicationHeader.java:451]: Nitpick: Braces on separate line for C++/Java.

**Review by gtritchie — CHANGES_REQUESTED** [review]
Source code style tweaks needed but otherwise looks good!

File: src/cpp/session/modules/SessionUpdates.cpp (hunk 2 above)
Comment (sharon-wang) [reply src/cpp/session/modules/SessionUpdates.cpp:137]: thanks Gary! I'll do a style guidelines pass for future contributions 👍

**Review by gtritchie — APPROVED** [review]
LGTM!

**Review by timtmok — APPROVED** [review]
I'm good with this pending the translation confirmation

File: src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants_fr.properties (hunk 1 above)
Comment (romainfrancois) [reply src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants_fr.properties:157]: ✅ 🇫🇷

**Pull request merged**